pkg/controller/registry/reconciler/grpc.go

```diff
@@ -252,64 +240,71 @@ func (c *GrpcRegistryReconciler) EnsureRegistryServer(logger *logrus.Entry, cata
 	// if service status is nil, we force create every object to ensure they're created the first time
 	valid, err := isRegistryServiceStatusValid(&source)
 	if err != nil {
+		logger.WithError(err).Error("error ensuring registry server: could not validate registry service status")
 		return err
 	}
 	overwrite := !valid
-	if overwrite {
-		logger.Info("registry service status invalid, need to overwrite")
-	}
 
 	//TODO: if any of these error out, we should write a status back (possibly set RegistryServiceStatus to nil so they get recreated)
 	sa, err := c.ensureSA(source)
 	if err != nil && !apierrors.IsAlreadyExists(err) {
+		logger.WithError(err).Error("error ensuring registry server: could not ensure registry service account")
 		return pkgerrors.Wrapf(err, "error ensuring service account: %s", source.GetName())
 	}
 
 	sa, err = c.OpClient.GetServiceAccount(sa.GetNamespace(), sa.GetName())
 	if err != nil {
+		logger.WithError(err).Error("error ensuring registry server: could not get registry service account")
 		return err
 	}
 
 	defaultPodSecurityConfig, err := getDefaultPodContextConfig(c.OpClient, catalogSource.GetNamespace())
 	if err != nil {
+		logger.WithError(err).Error("error ensuring registry server: could not get default pod security config")
 		return err
 	}
 
 	// recreate the pod if no existing pod is serving the latest image or correct spec
 	current, err := c.currentPodsWithCorrectImageAndSpec(logger, source, sa, defaultPodSecurityConfig)
 	if err != nil {
+		logger.WithError(err).Error("error ensuring registry server: could not get current pods with correct image and spec")
 		return err
 	}
 	overwritePod := overwrite || len(current) == 0
-	if overwritePod {
-		logger.Info("registry pods invalid, need to overwrite")
-	}
```

Comment:
Is there some other logging of the fact that we are going to override the pod (or at least some logging that only happens if we overwrite the pod), even in the happy path case?